indent/cylc.vim

```diff
@@ -0,0 +1,68 @@
+" Only load this indent file when no other was loaded.
+if exists("b:did_indent")
+  finish
+endif
+let b:did_indent = 1
+
+" Some preliminary settings
+setlocal nolisp      " Make sure lisp indenting doesn't supersede us
+setlocal autoindent  " indentexpr isn't much help otherwise
+
+setlocal indentexpr=GetCylcIndent(v:lnum)
+setlocal indentkeys=!^F,o,O,]
+
+" Only define the function once.
+if exists("*GetCylcIndent")
+  finish
+endif
+
+let s:save_cpo = &cpo
+set cpo&vim
+
+function s:header_level(lnum)
+  " Level of the section header on a given line, 0 if none
+  let line = getline(a:lnum)
+  return substitute(line, '^\s*\(\[*\).*', '\=strlen(submatch(1))', '')
+endfunction
+
+function s:prev_header(lnum)
+  " Jump to the header preceding a given line
+  call cursor(a:lnum, 1)
+  return search('^\s*\[\+', "bW")
+endfunction
+
+function GetCylcIndent(lnum)
+  let line = getline(a:lnum)
+  let line = substitute(line, '\v^\s+|\s+$', '', 'g')
+
+  " Do not adjust jinja indentation
+  if line =~ '^\({{|{%\)'
+    return -1
+  endif
+
+  " Check for a section header on this line
+  let level = s:header_level(a:lnum)
+  if level > 0
+    return shiftwidth() * (level - 1)
+  endif
+
+  " Otherwise, find the nearest section header
+  let level = s:header_level(s:prev_header(a:lnum))
+
+  " Add an extra indent inside multiline strings, not including closing
+  " quotes if on their own line
+  if has('syntax_items')
+    if synIDattr(synID(a:lnum, 1, 1), "name") =~ "String"
+      if line != '"""'
+        let level = level + 1
+      endif
+    endif
+  endif
```

Comment:
This bit doesn't seem to be working for me:

```ini
[scheduling]
    [[graph]]
            R1 = """
        # <- next line starts here       
```

I tried messing with this line:

```diff
-       let level = level + 1
+       let level = level + 10
```

But it didn't seem to make a difference, so I think it's failing to detect the multi-line string. Possibly a result of recent-ish changes to the syntax file?

Reply:
I've managed to narrow it down to this line:
```vim
    if synIDattr(synID(a:lnum, 1, 1), "name") =~ "String"
```

Which would appear to be evaluating as `trailingWS` rather than `cylcMultiString`.

Reply:
Fixed by considering the entire stack of applicable syntax items (`synID` -> `synstack`)